apps/condo/domains/acquiring/schema/MultiPayment.js

```diff
@@ -150,20 +189,118 @@ const MultiPayment = new GQLListSchema('MultiPayment', {
+                    id_in: payments.map(payment => payment.context),
+                })
+                const integrations = new Set(contexts.map(context => context.integration))
+                if (integrations.size !== 1) {
+                    return addValidationError(MULTIPAYMENT_MULTIPLE_ACQUIRING_INTEGRATIONS)
+                }
+                const integrationId = contexts[0].integration
+                if (resolvedData['integration'] !== integrationId) {
+                    return addValidationError(MULTIPAYMENT_ACQUIRING_INTEGRATIONS_MISMATCH)
+                }
+                const integration = await getById('AcquiringIntegration', integrationId)
+                if (!integration.canGroupReceipts && payments.length > 1) {
+                    return addValidationError(MULTIPAYMENT_CANNOT_GROUP_RECEIPTS)
+                }
+            } else if (operation === 'update') {
+                const payments = await find('Payment', {
+                    multiPayment: { id: existingItem.id },
+                })
+                const oldStatus = existingItem.status
+                const newStatus = get(resolvedData, 'status', oldStatus)
+                if (!MULTIPAYMENT_TRANSITIONS[oldStatus].includes(newStatus)) {
+                    return addValidationError(`${MULTIPAYMENT_NOT_ALLOWED_TRANSITION} Cannot move from "${oldStatus}" status to "${newStatus}"`)
+                }
+                const newItem = {
+                    ...existingItem,
+                    ...resolvedData,
+                }
+                const requiredFields = MULTIPAYMENT_REQUIRED_FIELDS[newStatus]
+                let requiredMissing = false
+                for (const field of requiredFields) {
+                    if (!newItem.hasOwnProperty(field) || newItem[field] === null) {
+                        addValidationError(`${MULTIPAYMENT_MISSING_REQUIRED_FIELDS} Field ${field} was not provided`)
+                        requiredMissing = true
+                    }
+                }
+                if (requiredMissing) return
+                const frozenFields = MULTIPAYMENT_FROZEN_FIELDS[oldStatus]
+                let frozenIncluded = false
+                for (const field of frozenFields) {
+                    if (resolvedData.hasOwnProperty(field)) {
+                        addValidationError(`${MULTIPAYMENT_FROZEN_FIELD_INCLUDED} (${field})`)
+                        frozenIncluded = true
+                    }
+                }
+                if (frozenIncluded) return
+                if (newStatus === MULTIPAYMENT_DONE_STATUS) {
+                    const undonePayments = payments
+                        .filter(payment => payment.status !== PAYMENT_DONE_STATUS)
+                        .map(payment => payment.id)
+                    if (undonePayments.length) {
+                        addValidationError(`${MULTIPAYMENT_UNDONE_PAYMENTS} Undone payments ids: ${undonePayments.join(', ')}`)
+                    }
```

Comment:
return ?

Reply:
Return is only needed if its absence interferes with the execution of logic from below. 
In this case payments statuses doesn't impact on logic below, so...no return needed

For more details read comment about stacking errors:
https://github.com/open-condo-software/condo/pull/807#discussion_r752110168